source/simulator/assemblers/newton_stokes.cc

```diff
@@ -687,6 +822,58 @@ namespace aspect
       const unsigned int stokes_dofs_per_cell = data.local_dof_indices.size();
       const unsigned int n_q_points = scratch.finite_element_values.n_quadrature_points;
       const double derivative_scaling_factor = this->get_newton_handler().parameters.newton_derivative_scaling_factor;
+      const double pressure_scaling = this->get_pressure_scaling();
+
+      const MaterialModel::MaterialAveraging::AveragingOperation
+      material_averaging = this->get_parameters().material_averaging;
+
+      // If elasticity is enabled, then we need ElasticOutputs to get the viscoelastic
+      // strain rate.
+      const std::shared_ptr<const MaterialModel::ElasticOutputs<dim>> elastic_out =
+        scratch.material_model_outputs.template get_additional_output_object<MaterialModel::ElasticOutputs<dim>>();
+
+      const std::shared_ptr<MaterialModel::MaterialModelDerivatives<dim>> derivatives
+        = scratch.material_model_outputs.template get_additional_output_object<MaterialModel::MaterialModelDerivatives<dim>>();
+
+      std::vector<double> deta_deps_times_grads_phi_u(stokes_dofs_per_cell);
+      std::vector<double> deta_dp_times_phi_p(stokes_dofs_per_cell);
+      std::vector<double> div_u_times_div_phi_u(stokes_dofs_per_cell);
```

Comment:
If I see this right, you only use these variables when you do averaging, right? If so, perhaps declare these variables here as empty arrays, and then resize them to the correct size inside the `if` condition immediately below -- this way, you don't pay the price of memory allocation if you don't actually need the variables.